workflows/prognostic_c48_run/runtime/transformers/core.py

```diff
@@ -68,7 +68,9 @@ def inputs_to_save(self) -> Set[str]:
         return set(strip_prefix(self.prefix, self.diagnostic_variables))
 
     def transform(self, func: Step) -> Diagnostics:
-        inputs: State = {key: self.state[key] for key in self.model.input_variables}
+        inputs: State = {
+            key: self.state[key].copy(deep=True) for key in self.model.input_variables
```

Comment:
Can you add a comment explaining why deep is needed? I'm assuming it's bc state[key] is an xarray object.

Reply:
Sure. And yes that's right.